Task3/EEL_task_3/src/main/java/com/epam/esm/exceptionhandler/handler/RestExceptionHandler.java

```diff
@@ -13,37 +13,54 @@
     }
 
     @ExceptionHandler(ObjectAlreadyExistsException.class)
-    public ResponseEntity<Object> handleObjectAlreadyExists(RuntimeException ex, WebRequest request) {
+    public ResponseEntity<?> handleObjectAlreadyExists(RuntimeException ex, WebRequest request) {
         return handleExceptionInternal(ex, ex.getMessage(), new HttpHeaders(), HttpStatus.CONFLICT, request);
     }
 
     @ExceptionHandler(ObjectIsInvalidException.class)
-    public ResponseEntity<Object> handleObjectIsInvalid(RuntimeException ex, WebRequest request) {
+    public ResponseEntity<?> handleObjectIsInvalid(RuntimeException ex, WebRequest request) {
         return handleExceptionInternal(ex, ex.getMessage(), new HttpHeaders(), HttpStatus.BAD_REQUEST, request);
     }
 
-    @ExceptionHandler(PageException.class)
-    public ResponseEntity<Object> handlePageException(RuntimeException ex, WebRequest request) {
-        return handleExceptionInternal(ex, ex.getMessage(), new HttpHeaders(), HttpStatus.NOT_FOUND, request);
+    @ExceptionHandler({JsonPatchException.class, JsonProcessingException.class})
+    public ResponseEntity<?> handleJsonException(RuntimeException ex, WebRequest request) {
+        return handleExceptionInternal(ex, CHECK_YOUR_REQUEST, new HttpHeaders(), HttpStatus.INTERNAL_SERVER_ERROR,
+                request);
     }
 
-    @ExceptionHandler(ObjectNotFoundException.class)
-    public ResponseEntity<Object> handleObjectNotFoundException(RuntimeException ex, WebRequest request) {
-        return handleExceptionInternal(ex, ex.getMessage(), new HttpHeaders(), HttpStatus.NOT_FOUND, request);
+    @ExceptionHandler(UserCreationFailureException.class)
+    public ResponseEntity<?> handleUserCreationException(RuntimeException ex, WebRequest request) {
+        return handleExceptionInternal(ex, ex.getMessage(), new HttpHeaders(), HttpStatus.CONFLICT,
+                request);
     }
 
-    @ExceptionHandler({JsonPatchException.class, JsonProcessingException.class})
-    public ResponseEntity<Object> handleJson(RuntimeException ex, WebRequest request) {
-        return handleExceptionInternal(ex, "An error occurred, check your request", new HttpHeaders(), HttpStatus.INTERNAL_SERVER_ERROR,
-                request);
+    @ExceptionHandler(EmailNotFoundException.class)
+    public ResponseEntity<?> emailNotFoundException(RuntimeException ex, WebRequest request) {
+        return handleExceptionInternal(ex, ex.getMessage(), new HttpHeaders(), HttpStatus.BAD_REQUEST, request);
+    }
+
+    @ExceptionHandler(InvalidTokenException.class)
+    public ResponseEntity<?> invalidTokenException(RuntimeException ex, WebRequest request) {
+        return handleExceptionInternal(ex, ex.getMessage(), new HttpHeaders(), HttpStatus.FORBIDDEN, request);
     }
+
+    @ExceptionHandler(WrongAuthenticationInstanceException.class)
+    public ResponseEntity<?> wrongInstanceDetectedException(RuntimeException ex, WebRequest request) {
```

Comment:
Why do you specify `?` in Response entity?

Reply:
Because I don't want to use Raw types and want to make response body more flexible. Should I use Object or something else instead?

Reply:
It's better to use model for error as often you need to provide contract(generated Swagger or OpenApi specification) for errors for consumers